### PR TITLE
Add "type": "module"

### PR DESCRIPTION
**Pull request by benmccann**
This will help add SvelteKit / Vite support (https://github.com/apollographql/apollo-client/issues/8218)

Closes https://github.com/apollographql/invariant-packages/pull/240
Closes https://github.com/apollographql/invariant-packages/issues/227

### CODE REVIEW

**Comment by benmccann** [comment]
I've signed the CLA. Thanks

**Comment by benmccann** [comment]
@benjamn thanks for approving the workflow. It looks like the tests were failing since adding `"type": "module"` switched them to run as ESM. I've updated it so that the tests pass now if you're able to kick off the workflow again. Thanks!

File: package.json [hunk 1]
```diff
@@ -25,9 +25,8 @@
     "@types/mocha": "^9.0.0",
     "@types/node": "^16.3.1",
     "lerna": "^4.0.0",
-    "mocha": "^9.0.2",
+    "mocha": "^9.2.0",
```
Comment (benjamn) [inline package.json:28]: Nice catch that `mocha@latest` supports ESM syntax!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node_version: ['10', '12', '14', '15', '16']
+        node_version: ['12', '14', '15', '16', '17']
```
Comment (benmccann) [inline .github/workflows/main.yml:15]: maybe drop 15 too?

**Review by benjamn — APPROVED** [review]
Thanks so much for this @benmccann!

File: .github/workflows/main.yml (hunk 2 above)
Comment (benjamn) [reply .github/workflows/main.yml:15]: I wish Node.js had usage statistics somewhere, but I think I'd like to keep v15 for now.

**Pull request merged**